From patchwork Tue Sep  4 15:57:32 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Yang <richard.weiyang@gmail.com>
X-Patchwork-Id: 10587563
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A9FD7112B
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Sep 2018 15:57:43 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 9CB26293F3
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Sep 2018 15:57:43 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 910D129D4E; Tue,  4 Sep 2018 15:57:43 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 47156293F3
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Sep 2018 15:57:43 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727497AbeIDUXY (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 4 Sep 2018 16:23:24 -0400
Received: from mail-pl1-f193.google.com ([209.85.214.193]:40915 "EHLO
        mail-pl1-f193.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726200AbeIDUXY (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 4 Sep 2018 16:23:24 -0400
Received: by mail-pl1-f193.google.com with SMTP id s17-v6so1831403plp.7
        for <kvm@vger.kernel.org>; Tue, 04 Sep 2018 08:57:41 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id;
        bh=/+z5m+qxGm4/VVbzTBIxQXOG7nWgAQDYpfbV3GEkrBI=;
        b=eR3wfGRypzYnJ78KtUA4f8G26sLQvgCT16CetqyS/64a9GYCMbWRxkp2a9MGmjK9he
         QOLMTCNAmdCO3tGeHs5VHKh6egFMFKib5SaRfz+BEc1hJ9dBgxrSTSU4dSRT9/zXAHIv
         kaMWIX116rpz6nbp/UJ/R521NwiwycsJGTVNfauEPM8mq087PsVY/rp1zh4wjsGAYKUw
         7jqGTCFz0v5uQ0ORfbhcRP/uIcWzuSPQIqZxgNN/2wSgYjEopf3gxwDjwHyWLwuBw+8m
         vo/5PoFRJAx2YqqL3/6aMGA2fgxV14uUWnlmrIbxdTfqW0R319FWPCQXruVkKHUGFZGE
         lPbQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=/+z5m+qxGm4/VVbzTBIxQXOG7nWgAQDYpfbV3GEkrBI=;
        b=tgNKMIeXkxLNTflKlL1W9ETCDqw5MeLCEKSPpiR77Kn+GLr23VHdnyWvuS+MB+QrWo
         2KrIHVYJlN5IiDs0E6KnLvNwikDqozjXvxxZeltiu1qIohuCJMOVyICMmrHK4z4Om04R
         cLwY/IEy+iq6qOLbC1K/UzggHB2+MUCoRJ3t5xPdl0KC11Wng06j8vSpqNtffjvNoR0f
         de9VoCDjxUPVpz0MKVSDkRbUVv1Sq8kJOsbnBqX8vfO/4VWtPOZr17rRyVEbhIELcXWu
         UgUw77hVPzfbb/vVpDLJCh7GCRuoytyzoxuEcEuFjXp5JQr1nquYwYfD4ZntBTDSH3wE
         3tqg==
X-Gm-Message-State: APzg51B+XUjEvjBGx1mu6EVeO7ZYTFqfZIPWGHXy6nOQB10OIgCiysOz
        8hY0zy9/2nyoHi2V44zvWI0=
X-Google-Smtp-Source: 
 ANB0VdZhrF1NLUtHXoKIuSSBa9wk0QMYUfNtcWqYPeoEHlgdHHGSKLKm/eQ6IF/0tGI94+v+Q+yzzg==
X-Received: by 2002:a17:902:b78a:: with SMTP id
 e10-v6mr34626362pls.79.1536076660991;
        Tue, 04 Sep 2018 08:57:40 -0700 (PDT)
Received: from localhost ([185.92.221.13])
        by smtp.gmail.com with ESMTPSA id
 t2-v6sm35935118pfj.7.2018.09.04.08.57.39
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Tue, 04 Sep 2018 08:57:40 -0700 (PDT)
From: Wei Yang <richard.weiyang@gmail.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com, tglx@linutronix.de
Cc: x86@kernel.org, kvm@vger.kernel.org,
        Wei Yang <richard.weiyang@gmail.com>
Subject: [PATCH] KVM: x86: return 0 in case kvm_mmu_memory_cache has min
 number of objects
Date: Tue,  4 Sep 2018 23:57:32 +0800
Message-Id: <20180904155732.2202-1-richard.weiyang@gmail.com>
X-Mailer: git-send-email 2.15.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The code tries to pre-allocate *min* number of objects, so it is ok to
return 0 when the kvm_mmu_memory_cache meets the requirement.

Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
---
 arch/x86/kvm/mmu.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/kvm/mmu.c b/arch/x86/kvm/mmu.c
index 28da8a9bd37b..aee296c579e1 100644
--- a/arch/x86/kvm/mmu.c
+++ b/arch/x86/kvm/mmu.c
@@ -916,7 +916,7 @@ static int mmu_topup_memory_cache(struct kvm_mmu_memory_cache *cache,
 	while (cache->nobjs < ARRAY_SIZE(cache->objects)) {
 		obj = kmem_cache_zalloc(base_cache, GFP_KERNEL);
 		if (!obj)
-			return -ENOMEM;
+			return (cache->nobjs >= min)?0 : -ENOMEM;
 		cache->objects[cache->nobjs++] = obj;
 	}
 	return 0;
@@ -944,7 +944,7 @@ static int mmu_topup_memory_cache_page(struct kvm_mmu_memory_cache *cache,
 	while (cache->nobjs < ARRAY_SIZE(cache->objects)) {
 		page = (void *)__get_free_page(GFP_KERNEL_ACCOUNT);
 		if (!page)
-			return -ENOMEM;
+			return (cache->nobjs >= min)?0 : -ENOMEM;
 		cache->objects[cache->nobjs++] = page;
 	}
 	return 0;
